core/modules/user/user.admin.inc

```diff
@@ -900,3 +620,315 @@ function user_admin_role_delete_confirm_submit($form, &$form_state) {
+    ),
+  );
+  $form['email_canceled']['settings']['user_mail_status_canceled_subject'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Subject'),
+    '#default_value' => $mail_config->get('status_canceled_subject'),
+    '#maxlength' => 180,
+    '#element_validate' => array('token_element_validate'),
+    '#token_types' => array('user'),
+  );
+  $form['email_canceled']['settings']['user_mail_status_canceled_body'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Body'),
+    '#default_value' => $mail_config->get('status_canceled_body'),
+    '#rows' => 3,
+    '#element_validate' => array('token_element_validate'),
+    '#token_types' => array('user'),
+  );
+
+  $form['actions']['#type'] = 'actions';
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save configuration')
+  );
+  return $form;
+}
+
+/**
+ * Form submission handler for user_settings_email().
+ */
+function user_settings_email_submit($form, &$form_state) {
+  $values = $form_state['values'];
+
+  $config = config('system.core');
+  $config->set('user_mail_status_activated_notify', $values['user_mail_status_activated_notify']);
+  $config->set('user_mail_status_blocked_notify', $values['user_mail_status_blocked_notify']);
+  $config->set('user_mail_status_canceled_notify', $values['user_mail_status_canceled_notify']);
+  $config->save();
+
+  $config = config('user.mail');
+  $config->set('cancel_confirm_body', $values['user_mail_cancel_confirm_body']);
+  $config->set('cancel_confirm_subject', $values['user_mail_cancel_confirm_subject']);
+  $config->set('password_reset_body', $values['user_mail_password_reset_body']);
+  $config->set('password_reset_subject', $values['user_mail_password_reset_subject']);
+  $config->set('register_admin_created_body', $values['user_mail_register_admin_created_body']);
+  $config->set('register_admin_created_subject', $values['user_mail_register_admin_created_subject']);
+  $config->set('register_no_approval_required_body', $values['user_mail_register_no_approval_required_body']);
+  $config->set('register_no_approval_required_subject', $values['user_mail_register_no_approval_required_subject']);
+  $config->set('register_pending_approval_body', $values['user_mail_register_pending_approval_body']);
+  $config->set('register_pending_approval_subject', $values['user_mail_register_pending_approval_subject']);
+  $config->set('status_activated_body', $values['user_mail_status_activated_body']);
+  $config->set('status_activated_subject', $values['user_mail_status_activated_subject']);
+  $config->set('status_blocked_body', $values['user_mail_status_blocked_body']);
+  $config->set('status_blocked_subject', $values['user_mail_status_blocked_subject']);
+  $config->set('status_canceled_body', $values['user_mail_status_canceled_body']);
+  $config->set('status_canceled_subject', $values['user_mail_status_canceled_subject']);
+  $config->save();
+
+  // Clear the token cache.
+  token_cache_clear();
```

Comment:
I think the token cache was only needed because of User module's `[user:picture]` token being dependent on the user picture settings. We shouldn't need it here for the email form.